### PR TITLE
Added Javadoc

### PR DESCRIPTION
**Pull request by cowwoc**

### CODE REVIEW

**Comment by cowwoc** [comment]
Fixes #3900

**Comment by cowwoc** [comment]
I need help with the eca check.

I filled out the necessary forms but I am still getting this error: `Anonymous (<email-address>) is not covered by the necessary legal agreements in order to proceed.`

**Comment by jmcc0nn3ll** [comment]
https://www.eclipse.org/jetty/documentation/9.4.x/contributing-patches.html

You need to make sure you make the commit with -s as mentioned in that guide I linked, it will put an email address in your commit message that is used by the ECA to validate your commit against.

**Comment by cowwoc** [comment]
Good catch. I will resubmit shortly.

**Comment by cowwoc** [comment]
ECA validation fixed. Please merge.

File: jetty-server/src/main/java/org/eclipse/jetty/server/HttpConfiguration.java [hunk 1]
```diff
@@ -318,11 +318,21 @@ public boolean getSendXPoweredBy()
         return _sendXPoweredBy;
     }
 
+    /**
+     * Indicates if the {@code Date} header should be sent in responses.
```
Comment (joakime) [inline jetty-server/src/main/java/org/eclipse/jetty/server/HttpConfiguration.java]: This is the HTTP specified standard header `Date`.
From https://tools.ietf.org/html/rfc7231#section-7.1.1.2
Registered at https://www.iana.org/assignments/message-headers/message-headers.xhtml

Too many people will read this javadoc and assume that `Date` is optional, superfluous, and something that Jetty just decided to do.   

Please add the following ...

``` java
@see <a href="https://tools.ietf.org/html/rfc7231#section-7.1.1.2">HTTP/1.1 Standard Header: Date</a>
```

File: jetty-server/src/main/java/org/eclipse/jetty/server/HttpConfiguration.java [hunk 2]
```diff
@@ -318,11 +318,21 @@ public boolean getSendXPoweredBy()
...
+     * @param sendDateHeader true if the {@code Date} header should be sent in responses
+     */
     public void setSendDateHeader(boolean sendDateHeader)
     {
         _sendDateHeader = sendDateHeader;
     }
 
+    /**
+     * Indicates if the {@code Date} header will be sent in responses.
+     *
```
Comment (joakime) [inline jetty-server/src/main/java/org/eclipse/jetty/server/HttpConfiguration.java]: Some language that indicates that this has a default value of `true` is needed here.

Comment (cowwoc) [reply jetty-server/src/main/java/org/eclipse/jetty/server/HttpConfiguration.java]: Updated. Please review.

File: jetty-server/src/main/java/org/eclipse/jetty/server/HttpConfiguration.java (hunk 1 above)
Comment (cowwoc) [reply jetty-server/src/main/java/org/eclipse/jetty/server/HttpConfiguration.java]: Updated. Please review.

Comment (joakime) [reply jetty-server/src/main/java/org/eclipse/jetty/server/HttpConfiguration.java]: Cool. much improved.
Thanks.

File: jetty-server/src/main/java/org/eclipse/jetty/server/HttpConfiguration.java (hunk 2 above)
Comment (joakime) [reply jetty-server/src/main/java/org/eclipse/jetty/server/HttpConfiguration.java]: Meh, it works for me. :-)

**Pull request merged**